### ISSUE TITLE
Connect-MgGraph doesn't work and doesn't report any information

### ISSUE THREAD

**Issue by chscott** [microsoftgraph/msgraph-sdk-powershell#496]
```powershell
C:\Temp> Connect-MgGraph -ClientId <REDACTED> -TenantId <REDACTED> -CertificateThumbprint <REDACTED>
C:\Temp> Get-MgContext
C:\Temp> Get-MgUser
Get-MgUser_List1: Authentication needed, call Connect-Graph.
```

I don't know what I'm doing wrong that prevents authentication from occurring, and surely there should be some feedback before Get-MgUser that indicates there's a problem. I'm following this guide: https://docs.microsoft.com/en-us/graph/powershell/app-only.
 
[AB#7383](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/7383)

**Comment by YuriySamorodov** [comment]
Same issue Microsoft Graph 1.2.0

**Comment by YuriySamorodov** [comment]
@chscott, please kindly make sure your certificate is in personal store. Looks like `Connect-MgGraph` does not check LocalMachine cert store.
Still looks like a bug

**Comment by peetrike** [comment]
I can also confirm that it doesn't work with certificate.  At the same time it works with delegated authentication in browser

**Comment by YuriySamorodov** [comment]
@peetrike , please make sure the cert is in your personal store cert:\Personal\My, not cert:\LocalMachine\My

**Comment by eckermike** [comment]
Requiring it be in teh personal store makes automation with GMSA's difficult.

**Comment by peetrike** [comment]
> @peetrike , please make sure the cert is in your personal store cert:\Personal\My, not cert:\LocalMachine\My

It is in personal store

**Comment by YuriySamorodov** [comment]
Lookks like something has been fixed. A code below works for me:

```
$TenantId = "some-tenant-id"
$ClientId = "some-client-id"
$Thumbprint = ‘cert thumbprint’
$localMachineCert = Get-ChildItem "Cert:\LocalMachine\$Thumbprint"
Connect-MgGraph -Certificate $localMachineCert -TenantId $TenantId -ClientId $ClientId
```

**Comment by peetrike** [comment]
There is still only version 1.2.0 on [PowerShellGallery.com](https://www.powershellgallery.com/packages?q=microsoft.graph).  Which version You use?

**Comment by peetrike** [comment]
The only way I get connection is using UserParameterSet:
`Connect-MgGraph -Scopes`, but as soon as I add `-TenantId` here, it stops working.

Unfortunately, UserParameterSet requires attended authentication, which means that it doesn't help running scripts unattended ...

**Comment by YuriySamorodov** [comment]
It's 1.2.0 as well.
Have you tried to recreate a certificate?
```
New-SelfSignedCertificate -DnsName veeamsoftwarecorp.onmicrosoft.com -FriendlyName "Microsoft Graph SDK" -CertStoreLocation cert:\CurrentUser\My
```

**Comment by peetrike** [comment]
Just tried with [MSAL.PL](https://github.com/jasoth/MSAL.PS) module and my certificate works also there.

It would be nice, if `Connect-MgGraph` could write to verbose or debug stream, to help understand, what exactly goes wrong (without generating any errors)

**Comment by peombwa** [comment]
This should no longer occur. Silent failure when using `Connect-MgGraph` was [fixed in v1.3.0](https://github.com/microsoftgraph/msgraph-sdk-powershell/releases/tag/1.3.0). The command should now write an error message to error stream when an exception is encountered.

As, @YuriySamorodov stated, you can use `-Certificate` parameter to load a certificate from any store/path like this:
``` powershell
#Option 1: Load from local machine store.
$cert = Get-ChildItem Cert:\LocalMachine\My\$CertThumbprint
Connect-Graph -Certificate $cert -TenantId $TenantId -ClientId $ClientId

# Option 2: Load from path.
$password= "SomePassword"
$passwordSecure = ConvertTo-SecureString -AsPlainText -Force $password
$cert = New-Object System.Security.Cryptography.X509Certificates.X509Certificate2("C:\Certificates\mycert.pfx", $passwordSecure)

Connect-Graph -Certificate $cert -TenantId $TenantId -ClientId $ClientId
```

**Comment by peombwa** [comment]
Feel free to reopen the issue is it still reoccurs.
